### PR TITLE
17 Fix null comparisons

### PR DESCRIPTION
**Pull request by Zejnilovic**

### CODE REVIEW

**Review by HuvarVer — APPROVED** [review]
Looks good. 
But there are missing more comprehensive tests.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/CliOptions.scala [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    implicit val formats: DefaultFormats.type = DefaultFormats
+
+    val fileStream = getClass.getResourceAsStream("/cli_options.json")
+    val jsonString = Source.fromInputStream(fileStream).mkString
+    val json = net.liftweb.json.parse(jsonString)
+    println(json.extract[CliHelp])
+  }
+
+  def parse(args: Array[String]): CliOptions = {
+    if (args.indexOf("--help") >= 0) {
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/CliOptions.scala:24]:
```suggestion
    if (args.contains("--help")) {
```

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+object DataframeOptions {
+  def validateAndCreate(options: Map[String, String]): DataframeOptions = {
+    val format = options.getOrElse("format", throw new MissingArgumentException(
+      """Format is mandatory option. Use
+        |"--format", "--ref-format" or "--new-format".""".stripMargin))
+    val path = if (format == "jdbc") {
+      options.getOrElse("dbtable", throw new MissingArgumentException(
+        """DB table name is mandatory option for format
+          |type jdbc. Use "--dbtable", "--ref-dbtable" or "--new-dbtable".""".stripMargin))
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:30]: Multiline string formatting:
```suggestion
        """DB table name is mandatory option for format
          |type jdbc. Use "--dbtable", "--ref-dbtable" or "--new-dbtable".
          |""".stripMargin))
```

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala [hunk 3]
```diff
@@ -0,0 +1,39 @@
+package za.co.absa.hermes.datasetComparison
+
+import org.apache.commons.cli.MissingArgumentException
+import org.apache.spark.sql._
+
+case class DataframeOptions(format: String, options: Map[String, String], path: String) {
+
+  private def setOptions(dfReader: DataFrameReader): DataFrameReader =
+    if ( options.isEmpty ) dfReader else dfReader.options(options)
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:9]: nit: extra white space inside parenthesis

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+  private def load(dfReader: DataFrameReader): DataFrame =
+    if (format == "jdbc") { dfReader.load()     }
+    else                  { dfReader.load(path) }
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:13]: nit: avoid vertical indentation.

Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DataframeOptions.scala:13]: nit: Redundant curly braces in one liner expressions.

**Comment by Zejnilovic** [comment]
> Looks good.
> But there are missing more comprehensive tests.

That is from the JDBC PR. I will create issue.

File: datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala [hunk 5]
```diff
@@ -32,44 +30,36 @@ object DatasetComparisonJob {
...
   def main(args: Array[String]): Unit = {
-    val cmd = DatasetComparisonConfig.getCmdLineArguments(args)  match {
-      case Success(value) => value
-      case Failure(exception) => throw exception
-    }
+    val cliOptions = CliOptions.parse(args)
 
     implicit val sparkSession: SparkSession = SparkSession.builder()
-      .appName(s"Dataset comparison - '${cmd.newPath}' and '${cmd.refPath}'")
+      .appName(s"Dataset comparison - '${cliOptions.referenceOptions.path}' and " +
```
Comment (wajda) [inline datasetComparison/src/main/scala/za/co/absa/hermes/datasetComparison/DatasetComparisonJob.scala:36]: string concat

**Pull request merged**